CMakeLists.txt

```diff
@@ -109,77 +112,74 @@ set(MWENGINE_PROCESSORS ${CPP_SRC}/processors/bitcrusher.cpp
 set(JNI_SOURCES ${CPP_SRC}/jni/javabridge.cpp
                 ${CPP_SRC}/jni/javautilities.cpp)
 
-if (NOT ${RUN_TESTS} MATCHES "true")
-
-    add_library(mwengine STATIC
-                         ${MWENGINE_CORE_SOURCES}
-                         ${MWENGINE_SYNTH_SOURCES}
-                         ${MWENGINE_PROCESSORS}
-                         ${JNI_SOURCES})
+# build the mock driver into the library when unit test execution is enabled
 
-    # link all the NDK libraries used by the project
-
-    target_link_libraries(mwengine android atomic log OpenSLES aaudio)
+if (${RUN_TESTS} MATCHES "true")
+    SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D=MOCK_ENGINE")
+    set(MWENGINE_CORE_SOURCES ${MWENGINE_CORE_SOURCES}
+                              ${CPP_SRC}/drivers/mock_io.cpp)
+endif()
```

Comment:
Mock driver code is only compiled when the test runner is configured for the build.